File: modules/graphics/lib/rin/graphics/SamplerSpec.cpp
#include "rin/graphics/SamplerSpec.hpp"

SamplerSpec::SamplerSpec(const vk::Filter& inFilter, const vk::SamplerAddressMode& inTiling)
{
    filter = inFilter;
    tiling = inTiling;
}

std::string SamplerSpec::GetId() const
{
    return std::to_string(static_cast<int>(filter)) + "_" + std::to_string(static_cast<int>(tiling));
}

File: rin.Graphics/SamplerSpec.cs
using TerraFX.Interop.Vulkan;

namespace rin.Graphics;

public struct SamplerSpec
{
    public ImageTiling Tiling;
    public ImageFilter Filter;

    public static implicit operator VkSampler(SamplerSpec spec)
    {
        return SGraphicsModule.Get().GetSampler(spec);
    }
}